src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -274,7 +279,11 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     void TerminalTab::SplitPane(SplitState splitType, const GUID& profile, TermControl& control)
     {
+        const auto activePaneId = _activePane->PaneId();
         auto [first, second] = _activePane->Split(splitType, profile, control);
+        first->PaneId(activePaneId);
+        second->PaneId(_nextPaneId);

```

Comment:
So the Tab is responsible for assigning IDs here as panes get split, that makes sense.

However, what happens when I make two panes `[ 1 | 2 ]`, then close pane 1. The parent pane will absorb the control from pane 2 into itself, but won't that pane still have the ID `1`? 

Should the Pane make sure to clear out it's assigned ID when a split is being made, and also make sure to absorb the ID of the remaining child (if needed)?

Reply:
Thanks for the catch! We absorb the remaining child's ID and clear out ID when splitting now